internal/util/duration.go

```diff
@@ -0,0 +1,105 @@
+package util
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Common duration constants for convenience
+const (
+	Day  = 24 * time.Hour
```

Comment:
Are we OK that a day might not always equal to 24 hours?
Example: Daylight savings (23 or 25), leap seconds